From 4c5d336b44bc6122a6721905e77849ebd57eca78 Mon Sep 17 00:00:00 2001
From: Philip Chimento <philip.chimento@gmail.com>
Date: Wed, 5 Jul 2017 22:53:30 -0700
Subject: [PATCH] build: Remove unnecessary NSPR dependency

When SpiderMonkey is configured with --enable-posix-nspr-emulation, then
the dependency on NSPR can be removed from the pkg-config file.

https://bugzilla.mozilla.org/show_bug.cgi?id=1379539
---
 build/autoconf/nspr-build.m4 | 11 ++++++-----
 js/src/js.pc.in              |  2 +-
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/build/autoconf/nspr-build.m4 b/build/autoconf/nspr-build.m4
index fe6dab72..b733579c 100644
--- a/build/autoconf/nspr-build.m4
+++ b/build/autoconf/nspr-build.m4
@@ -177,11 +177,8 @@ AC_SUBST_LIST(NSPR_CFLAGS)
 AC_SUBST(NSPR_INCLUDE_DIR)
 AC_SUBST(NSPR_LIB_DIR)
 
-NSPR_PKGCONF_CHECK="nspr"
+PKGCONF_REQUIRES_PRIVATE="Requires.private: nspr"
 if test -n "$MOZ_SYSTEM_NSPR"; then
-    # piggy back on $MOZ_SYSTEM_NSPR to set a variable for the nspr check for js.pc
-    NSPR_PKGCONF_CHECK="nspr >= $NSPR_MINVER"
-
     _SAVE_CFLAGS=$CFLAGS
     CFLAGS="$CFLAGS $NSPR_CFLAGS"
     AC_TRY_COMPILE([#include "prlog.h"],
@@ -191,8 +188,12 @@ if test -n "$MOZ_SYSTEM_NSPR"; then
                 ,
                 AC_MSG_ERROR([system NSPR does not support PR_STATIC_ASSERT]))
     CFLAGS=$_SAVE_CFLAGS
+    # piggy back on $MOZ_SYSTEM_NSPR to set a variable for the nspr check for js.pc
+    PKGCONF_REQUIRES_PRIVATE="Requires.private: nspr >= $NSPR_MINVER"
+elif test -n "$JS_POSIX_NSPR"; then
+    PKGCONF_REQUIRES_PRIVATE=
 fi
-AC_SUBST(NSPR_PKGCONF_CHECK)
+AC_SUBST([PKGCONF_REQUIRES_PRIVATE])
 
 fi # _IS_OUTER_CONFIGURE
 
diff --git a/js/src/js.pc.in b/js/src/js.pc.in
index 1efea334..2eae393a 100644
--- a/js/src/js.pc.in
+++ b/js/src/js.pc.in
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: SpiderMonkey @MOZILLA_VERSION@
 Description: The Mozilla library for JavaScript
 Version: @MOZILLA_VERSION@
-Requires.private: @NSPR_PKGCONF_CHECK@
+@PKGCONF_REQUIRES_PRIVATE@
 Libs: -L${libdir} -l@JS_LIBRARY_NAME@
 Cflags: -include ${includedir}/@JS_LIBRARY_NAME@/js/RequiredDefines.h -I${includedir}/@JS_LIBRARY_NAME@

